src/openforms/js/components/formio_builder/WebformBuilder.js

```diff
@@ -184,6 +184,13 @@ class WebformBuilder extends WebformBuilderFormio {
         // Clean up.
         this.removeEventListener(this.dialog, 'close', dialogClose);
         this.dialog = null;
+
+        // We have to reset the ckeditor BodyCollection to allow link popups to work
+        // after closing the editComponent modal, because it seems that `ClassicEditor.destroy` isn't being called
+        // https://ckeditor.com/docs/ckeditor5/latest/api/module_ui_editorui_bodycollection-BodyCollection.html
+        // See https://github.com/open-formulieren/open-forms/issues/4012
+        if (document.querySelector('.ck-body-wrapper'))
+          document.querySelector('.ck-body-wrapper').remove();
```

Comment:
I was curious - perhaps the problem is that `root.unmount()` is not called, and therefore, the components cleanup functions from useEffect are not being called? This would effectively mean we are currently memory-leaking too :thinking: 

As I understand it, the the modal close by clicking outside of it or on the X basically removes that whole DOM node and since Formio is not aware of our react stuff, it can't properly clean that up.

I'd like to validate that theory before accepting this workaround.

Reply:
You were right! Calling `root.unmount` fixes in `dialogClose` fixes it

Reply:
:tada: thanks for doing the hard work and pointing us in the right direction!